MekHQ/src/mekhq/campaign/CampaignOptions.java

```diff
@@ -3852,14 +3852,14 @@ public static CampaignOptions generateCampaignOptionsFromXml(Node wn) {
             } else if (wn2.getNodeName().equalsIgnoreCase("intensity")) { // Legacy
                 double intensity = Double.parseDouble(wn2.getTextContent().trim());
 
-                retVal.atbBattleChance[AtBLanceRole.FIGHTING.ordinal()] = ((40.0 * intensity) / (40.0 * intensity + 60.0)) * 100.0 + 0.5;
-                retVal.atbBattleChance[AtBLanceRole.DEFENCE.ordinal()] = ((20.0 * intensity) / (20.0 * intensity + 80.0)) * 100.0 + 0.5;
-                retVal.atbBattleChance[AtBLanceRole.SCOUTING.ordinal()] = ((60.0 * intensity) / (60.0 * intensity + 40.0)) * 100.0 + 0.5;
-                retVal.atbBattleChance[AtBLanceRole.TRAINING.ordinal()] = ((10.0 * intensity) / (10.0 * intensity + 90.0)) * 100.0 + 0.5;
+                retVal.atbBattleChance[AtBLanceRole.FIGHTING.ordinal()] = (int) Math.round(((40.0 * intensity) / (40.0 * intensity + 60.0)) * 100.0 + 0.5);
+                retVal.atbBattleChance[AtBLanceRole.DEFENCE.ordinal()] = (int) Math.round(((20.0 * intensity) / (20.0 * intensity + 80.0)) * 100.0 + 0.5);
+                retVal.atbBattleChance[AtBLanceRole.SCOUTING.ordinal()] = (int) Math.round(((60.0 * intensity) / (60.0 * intensity + 40.0)) * 100.0 + 0.5);
+                retVal.atbBattleChance[AtBLanceRole.TRAINING.ordinal()] = (int) Math.round(((10.0 * intensity) / (10.0 * intensity + 90.0)) * 100.0 + 0.5);
             } else if (wn2.getNodeName().equalsIgnoreCase("atbBattleChance")) {
                 String[] values = wn2.getTextContent().split(",");
                 for (int i = 0; i < values.length; i++) {
-                    retVal.atbBattleChance[i] = Double.parseDouble(values[i]);
+                    retVal.atbBattleChance[i] = Integer.parseInt(values[i]);
```

Comment:
How will this handle older saves?

Reply:
Not been on any released versions, so I didn't think it would be necessary... however just remembered that a few outside of the dev team have been using this, so I'll add migration now.

Reply:
Implemented a (not great, but working) migration method. We want it to fail if it can't parse the boolean.